Google.Api.Gax.Grpc/GaxGrpcServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * Copyright 2022 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Api.Gax.Grpc;
+using Google.Api.Gax.Grpc.Rest;
+using Grpc.Net.Client;
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for dependency injection.
+    /// </summary>
+    public static class GaxGrpcServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds a singleton <see cref="GrpcNetClientAdapter"/> to the given service collection
+        /// as the preferred <see cref="GrpcAdapter"/> implementation,
+        /// using the default instance with any additional options configured via <paramref name="optionsConfigurer"/>.
+        /// </summary>
+        /// <param name="services">The service collection to add the adapter to.</param>
+        /// <param name="optionsConfigurer">The configuration action to perform on each <see cref="Grpc.Net.Client.GrpcChannelOptions"/>
+        /// when it is used by the adapter to construct a channel.</param>
+        /// <returns>The same service collection reference, for method chaining.</returns>
+        public static IServiceCollection AddGrpcNetClientAdapter(this IServiceCollection services, Action<IServiceProvider, Grpc.Net.Client.GrpcChannelOptions> optionsConfigurer) =>
+            services.AddSingleton<GrpcAdapter>(provider => GrpcNetClientAdapter.Default.WithAdditionalOptions(options => optionsConfigurer(provider, options)));
```

Comment:
Should't you add here the provider as well to the ChannelOptions, as there's no guarantee that the configurer action will do that.
Wether you do it before or after the configurer function executes, I'm fine with either, tending to doing it before to allow users to override that for some weird case.